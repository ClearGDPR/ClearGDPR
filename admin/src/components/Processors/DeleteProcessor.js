import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-responsive-modal';

import Loader from 'components/core/cards/dashboard/Loader';

const DeleteProcessor = props => (
  <Modal open={props.isOpen} showCloseIcon={false} onClose={() => {}} center>
    {!props.isLoading ? (
      <React.Fragment>
        <p>Are you sure you want to delete this processor?</p>
        <p>
          <button
            className="ui-action btn"
            onClick={e => {
              e.preventDefault();
              props.onCancel();
            }}
          >
            Cancel
          </button>
          <button
            className="ui-action btn"
            onClick={e => {
              e.preventDefault();
              props.onConfirm();
            }}
          >
            OK
          </button>
        </p>
      </React.Fragment>
    ) : (
      <Loader />
    )}
  </Modal>
);

DeleteProcessor.propTypes = {
  onConfirm: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  isOpen: PropTypes.bool,
  isLoading: PropTypes.bool
};

export default DeleteProcessor;
