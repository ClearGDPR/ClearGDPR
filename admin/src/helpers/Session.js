import cookie from 'cookie';

export class Session {
  set(authResult) {
    const maxAge = 1 * 24 * 60 * 60; // 1 day long;
    document.cookie = cookie.serialize('auth', authResult, { maxAge });
    document.cookie = cookie.serialize('expires_at', authResult, { maxAge });
  }

  destroy() {
    document.cookie = cookie.serialize('auth', '', { maxAge: -1 });
    document.cookie = cookie.serialize('expires_at', '', { maxAge: -1 });
  }

  isLoggedIn() {
    let expiresAt = cookie.parse(document.cookie).expires_at || '';
    return new Date().getTime() < expiresAt;
  }
}

export default new Session();
