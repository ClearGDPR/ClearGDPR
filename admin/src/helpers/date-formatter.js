import { format } from 'date-fns';

export default date => {
  return format(new Date(date), 'DD/MM/YYYY h:mma');
};
