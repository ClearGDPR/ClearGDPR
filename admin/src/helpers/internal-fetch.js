import merge from 'lodash/object/merge';
import session from '../../helpers/Session';

const internalFetch = (url, config = {}) => {
  const internalConfig = merge(
    {
      method: 'GET',
      headers: {
        Authorization: `Bearer ${session.getToken()}`,
        'Content-Type': 'application/json'
      }
    },
    config
  );
  return fetch(url, internalConfig).then(async res => {
    if (res.status === 400 && (await res.json()).error === 'JWT token expired') {
      session.destroy();
    } else {
      return res;
    }
  });
};

export default internalFetch;
