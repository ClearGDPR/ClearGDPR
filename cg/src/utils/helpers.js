const Promise = require('bluebird');
const winston = require('winston');

const { VALID_RUN_MODES } = require('./constants');

function inControllerMode() {
  return process.env.MODE === VALID_RUN_MODES.CONTROLLER;
}

function getMyAddress() {
  return process.env.MY_ADDRESS;
}
function getTokenExpiry() {
  return process.env.JWT_TOKEN_EXPIRY || 604800;
}

const timeout = ms => new Promise(res => setTimeout(res, ms));

async function retryAsync(promise, options = {}) {
  let tries = 0;

  const maxRetries = options.maxRetries || 3;
  const interval = options.interval || 500;
  let delay = 0;

  while (tries < maxRetries) {
    await Promise.delay(delay);
    tries++;
    try {
      return await promise();
    } catch (e) {
      delay = interval * tries;
      winston.info(`Attempt no. ${tries} failed. Next attempt in ${delay}ms`);
      winston.log('debug', `Error: ${e.toString()}`);
    }
  }

  winston.error(`Operation failed after ${tries} attempts`);
}

module.exports = {
  getTokenExpiry,
  inControllerMode,
  getMyAddress,
  timeout,
  retryAsync
};
