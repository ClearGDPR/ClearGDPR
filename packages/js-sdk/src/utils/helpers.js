import { merge } from 'lodash/object';

export function parseJSON(response) {
  const contentType = response.headers.get('Content-Type');

  if (contentType.indexOf('application/json') > -1) {
    return new Promise(resolve =>
      response.json().then(json => {
        return resolve({
          status: response.status,
          ok: response.ok,
          data: json
        });
      })
    );
  } else {
    return new Promise(resolve =>
      response.text().then(text => {
        return resolve({
          status: response.status,
          ok: response.ok,
          data: text
        });
      })
    );
  }
}

export function buildRequestBody(body) {
  return body ? JSON.stringify(body) : null;
}

/**
 * Merges objects values
 * @param {Object} options objects to be merged
 * @returns {Object} New merged object based on params
 */
export function mergeOptions(...objects) {
  return merge(...objects);
}
