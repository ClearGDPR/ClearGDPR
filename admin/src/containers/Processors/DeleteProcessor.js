import React from 'react';
import PropTypes from 'prop-types';

import { ProcessorsConsumer } from './ProcessorsContext';
import DeleteProcessor from 'components/Processor/DeleteProcessor';

export class DeleteProcessorContainer extends React.Component {
  static propTypes = {
    processor: PropTypes.object.isRequired,
    deleteProcessor: PropTypes.func.isRequired,
    onClose: PropTypes.func,
    isOpen: PropTypes.bool,
    isLoading: PropTypes.bool
  };

  deleteProcessor() {
    this.props.deleteProcessor(this.props.processor.id).then(this.props.onClose);
  }

  render() {
    return (
      <DeleteProcessor
        onConfirm={this.deleteProcessor.bind(this)}
        onCancel={this.props.onClose}
        isOpen={this.props.isOpen}
        isLoading={this.props.isLoading}
      />
    );
  }
}

export default props => (
  <ProcessorsConsumer>
    {({ deleteProcessor, isLoading }) => (
      <DeleteProcessorContainer {...props} isLoading={isLoading} deleteProcessor={deleteProcessor} />
    )}
  </ProcessorsConsumer>
);
