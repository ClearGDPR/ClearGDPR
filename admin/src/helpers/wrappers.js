import React from 'react';
import PropTypes from 'prop-types';
import lodash from 'lodash';
import { UsersProvider } from 'containers/Users/UsersContext';
import { ProcessorsProvider } from 'containers/Processors/ProcessorsContext';
import { PanelProvider } from 'containers/MainLayout/PanelContext';
import { RectificationsProvider } from 'containers/Rectifications/RectificationsContext';
import { StatsProvider } from 'containers/Dashboard/StatsContext';

export const withWrapper = Wrapper => Component => {
  let wrapper = props => (
    <Wrapper>
      <Component {...props}>{props.children}</Component>
    </Wrapper>
  );
  wrapper.propTypes = {
    children: PropTypes.node
  };
  return wrapper;
};

export const withContextProviders = lodash.flow.apply(null, [
  withWrapper(UsersProvider),
  withWrapper(ProcessorsProvider),
  withWrapper(RectificationsProvider),
  withWrapper(PanelProvider),
  withWrapper(StatsProvider)
]);
