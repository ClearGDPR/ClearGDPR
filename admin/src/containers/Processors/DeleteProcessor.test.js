import React from 'react';
import { shallow } from 'enzyme';

import * as TestUtils from 'tests/helpers/TestUtils';

import { DeleteProcessorContainer } from './DeleteProcessor';

const setup = propOverrides => {
  const props = Object.assign(
    {
      processorId: 1,
      deleteProcessor: jest.fn().mockReturnValue(Promise.resolve()),
      onClose: jest.fn(),
      isOpen: true,
      isLoading: false
    },
    propOverrides
  );
  const component = shallow(<DeleteProcessorContainer {...props} />);

  return { props, component };
};

beforeEach(() => {
  jest.clearAllMocks();
});

describe('(Container) Delete processor', () => {
  it('should render correctly when processor and default provided props provided', async () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });

  it('should call delete processor from props when delete processor was called', async () => {
    const { props, component } = setup();
    component.instance().deleteProcessor();
    expect(props.deleteProcessor).toHaveBeenCalledWith(1);
  });

  it('should call onClose when deleting processor is successful', async () => {
    const { props, component } = setup();
    component.instance().deleteProcessor();
    await TestUtils.flushPromises();
    expect(props.onClose).toHaveBeenCalled();
  });

  it('should call onClose when deleting processor is unsuccessful', async () => {
    const { props, component } = setup({
      deleteProcessor: jest.fn().mockReturnValue(Promise.reject('Some error!'))
    });
    await component.instance().deleteProcessor();
    await TestUtils.flushPromises();
    expect(props.onClose).toHaveBeenCalled();
  });
});
