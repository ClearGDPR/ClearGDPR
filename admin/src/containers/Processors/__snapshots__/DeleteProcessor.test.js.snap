// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) Delete processor should render correctly when processor and default provided props provided 1`] = `
<DeleteProcessor
  isLoading={false}
  isOpen={true}
  onCancel={[MockFunction]}
  onConfirm={[Function]}
/>
`;
